### PR TITLE
Fix a false positive with user-defined `Enum` class

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description

Closes PyCQA/pylint#5719

### CODE REVIEW

File: astroid/brain/brain_namedtuple_enum.py [hunk 1]
```diff
@@ -33,6 +33,8 @@
     "IntFlag",
     "enum.IntFlag",
 }
+ENUM_QNAME: Final = "enum.Enum"
```
Comment (DanielNoord) [inline astroid/brain/brain_namedtuple_enum.py]:
```suggestion
ENUM_QNAME: Final[str] = "enum.Enum"
```

File: astroid/brain/brain_namedtuple_enum.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import keyword
 from collections.abc import Iterator
 from textwrap import dedent
+from typing import Final
```
Comment (DanielNoord) [inline astroid/brain/brain_namedtuple_enum.py:13]: This needs a `sys` version check.

File: ChangeLog [hunk 3]
```diff
@@ -34,6 +34,11 @@ Release date: 2023-01-08
 
   Closes #1942
 
+* Fix a false positive when an attribute named ``Enum`` was confused with ``enum.Enum``.
```
Comment (DanielNoord) [inline ChangeLog:37]: This needs to be moved to `2.13.3`

**Review by DanielNoord — CHANGES_REQUESTED** [review]

File: ChangeLog [hunk 4]
```diff
@@ -34,6 +39,7 @@ Release date: 2023-01-08
 
   Closes #1942
 
+
```
Comment (mbyrnepr2) [inline ChangeLog:42]:
```suggestion
```

File: astroid/brain/brain_namedtuple_enum.py [hunk 5]
```diff
@@ -33,6 +39,8 @@
     "IntFlag",
     "enum.IntFlag",
 }
+ENUM_QNAME: Final[str] = "enum.Enum"
+TYPING_NAMEDTUPLE_BASENAMES = {"NamedTuple", "typing.NamedTuple"}
```
Comment (DanielNoord) [inline astroid/brain/brain_namedtuple_enum.py:43]:
```suggestion
TYPING_NAMEDTUPLE_BASENAMES: Final[set[str]] = {"NamedTuple", "typing.NamedTuple"}
```

File: astroid/brain/brain_namedtuple_enum.py [hunk 6]
```diff
@@ -298,6 +306,19 @@ def infer_enum(
     node: nodes.Call, context: InferenceContext | None = None
 ) -> Iterator[bases.Instance]:
     """Specific inference function for enum Call node."""
+
```
Comment (DanielNoord) [inline astroid/brain/brain_namedtuple_enum.py:309]:
```suggestion
```

File: tests/unittest_brain.py [hunk 7]
```diff
@@ -1230,6 +1230,42 @@ class MyEnum(PyEnum):
         assert isinstance(inferred, bases.Instance)
         assert inferred._proxied.name == "ENUM_KEY"
 
+    def test_class_named_enum(self) -> None:
+        """Test the user-defined class named `Enum` is not inferred as `enum.Enum`"""
```
Comment (DanielNoord) [inline tests/unittest_brain.py:1234]:
```suggestion
        """Test that the user-defined class named `Enum` is not inferred as `enum.Enum`"""
```

**Review by DanielNoord — CHANGES_REQUESTED** [review]
I like the test, very elegant!

Couple of nits, rest LGTM!

**Comment by mbyrnepr2** [comment]
Thanks for those suggestions @DanielNoord; nothing I can disagree with there!

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by DanielNoord — APPROVED** [review]
I'd like to see coverage, but other than that this LGTM!

**Comment by DanielNoord** [comment]
@mbyrnepr2 Do you think you can get the diff coverage to 100% here?

**Review by DanielNoord — APPROVED** [review]
Thanks for the additional test!

**Comment by DanielNoord** [comment]
@mbyrnepr2 Could you try and backport this? For some reason `git` doesn't recognize the `maintenance` branch for me locally.

**Comment by mbyrnepr2** [comment]
Done @DanielNoord: https://github.com/PyCQA/astroid/pull/1971. Let me know if any issues.

**Comment by DanielNoord** [comment]
Thanks! I feel bad for asking, but could you do https://github.com/PyCQA/astroid/pull/1970 as well? :smile:

It's my own PR but I haven't been able to fix the broken `git remote` issue on my local machine.. :sweat_smile:

**Pull request merged**